### ISSUE TITLE
[help] vault_pki_secret_backend_intermediate_set_signed wants to be recreated when upgrade from 3 to 4

### ISSUE THREAD

**Issue by zoezhangmattr** [hashicorp/terraform-provider-vault#2342]
### Description

hey there, trying to understand if there is any risk when doing version upgrade 3 to 4, and need some help or advice please.
i went through the upgrade guide https://registry.terraform.io/providers/hashicorp/vault/latest/docs/guides/version_4_upgrade

when i try to upgrade, do a `plan` 
the only diff thing is concerning is
vault_pki_secret_backend_intermediate_set_signed this resource, will need to be recreated, only diff i can see is it has 2 new fields
import_keys, and import_issuers

our pki is in use

i was wonder if its ok to apply the change, does it impact the current pki 

### Would you like to implement a fix?

None

**Comment by encbladexp** [comment]
We are facing the same issue.

**Comment by zoezhangmattr** [comment]
> Some update: It doesn't change/break anything, only recreates the resource with the same certificate within vault, so the provider's state contains these two new fields.
> 
> Reason seems to be that the API endpoint of Vault used, doesn't contain an ability to fetch data for these two fields.

hey there, thank you for your post, did you apply it without any impact in the production use?

**Comment by encbladexp** [comment]
> hey there, thank you for your post, did you apply it without any impact in the production use?

Exactly, as the signed certificate stays the same, it just does an reupload. Which then exports the additional attributes of the new provider version to the TF state.

The [docs](https://registry.terraform.io/providers/hashicorp/vault/latest/docs/guides/pki_multi_issuer_upgrade) contain important details:
> [vault_pki_secret_backend_intermediate_set_signed](https://registry.terraform.io/providers/hashicorp/vault/latest/docs/guides/pki_multi_issuer_upgrade#vault_pki_secret_backend_intermediate_set_signed-3): If any newly imported issuers are received from Vault, the old set-signed will be destroyed and a new one will be initiated.

If you are only having a change on set_signed, nothing bad will happen.
